roles/rsyslog/tasks/main.yaml

```diff
@@ -13,7 +13,7 @@
   set_fact:
     __rsyslog_version: >-
       {{ ansible_facts.packages['rsyslog'][0].version
-      if ansible_facts.packages | selectattr('rsyslog', 'defined') | list | length > 0
+      if ansible_facts.packages | json_query('rsyslog')
```

Comment:
> is there a test for the problem being fixed here? I am surprised that the problem was not caught before merging #87

I will add a test case.

Reply:
It does not seem like a good way to test this. It is an internal variable; what if it gets renamed or its scope restricted? This way you are basically relying on the unfortunate fact that variables leak out from the role. Can't you rather test for the actual functionality that needs this variable? I mean, before you fixed the problem, the role must have behaved differently in some situation, otherwise the whole code would be useless.

Reply:
@pcahyna, currently, the version info is used only when rsyslog_in_image is true, which is not supported in the test-harness, I'd think.  Do you have any recommendation to do in such a case?  As you described above, the assertion to check `__rsyslog_version != "0.0.0"` is not ideal, but probably we could detect an error if some mismatch was introduced in a new pull request?

Reply:
Ah. So there is even a test missing for `rsyslog_in_image`, so there is a bigger problem.